rdpng.c

```diff
@@ -102,8 +125,18 @@ start_input_png (j_compress_ptr cinfo, cjpeg_source_ptr sinfo)
     png_charp unused1 = NULL;
     int unused2 = 0;
     png_uint_32 proflen = 0;
-    if (png_get_iCCP(source->png_ptr, source->info_ptr, &unused1, &unused2, &profile, &proflen) && /* your libpng is out of date if you get a warning here */
-        profile && proflen) {
+    int has_profile = 0;
+
+    if (has_srgb_chunk) {
+        /* PNG can declare use of an sRGB profile without embedding an ICC file, but JPEG doesn't have such feature */
```

Comment:
Thanks. That other method requires parsing a EXIF, and Adobe's take on it is especially complex (with absolute offsets embedded in TIFF tags), so I'll pass.